### ISSUE TITLE
Improve use of title in footer

### ISSUE THREAD

**Issue by patcon** [GCTC-NTGC/gc-digital-talent#2902]
While converting `footer.spec.js` to use the testing-library's more a11y-minded testing-library selector `cy.findByRole()`, I realized that we were perhaps misusing (in this one place) how we use `title` attributes on links.

This is what we have now (once the cypress test PR is merged):
https://github.com/GCTC-NTGC/gc-digital-talent/blob/b373c91738f0976a45c4dbd2147082995e31d03c/frontend/cypress/integration/talentsearch/footer.spec.js#L2-L15

This is what we should be able to do:
```js
      // contentinfo is the aria role for <footer>
      cy.findByRole('contentinfo').within(() => {
        cy.findByRole('link', { name: 'Privacy Policy' }).should('have.attr', 'href', '/en/privacy-notice')
      })
```

But this isn't working, because name is resolved by testing-library (like a screenreader would) to use the `title` attribute on each footer link, which looks like this:

```html
<a href="#" title="Visit our privacy policy.">Privacy Policy</a>
```

This led me to deque's docs about how screenreaders treat this situation when anchor links are supplemented with `title` attributes: (spoiler: when different, thye read both aloud)
https://www.deque.com/blog/text-links-practices-screen-readers/

So for this screenreader is **reading out loud "link: privacy policy visit our privacy policy"**. (this seems a poor experience)

```html
<a href="#">Privacy Policy</a>
```

The above is best, and reads "link: privacy policy", as expected.

Proposal for here and anywhere else: **we shouldn't be using titles when link text already describes** what's behind it. Links are expected to, by default, "visit" the page.

(The place we might supplement a link with `title` attr is for things like a "mailto" href or [file download links](https://a11y-guidelines.orange.com/en/articles/download-links/).)

**Comment by mnigh** [comment]
> Proposal for here and anywhere else: we shouldn't be using titles when link text already describes what's behind it. Links are expected to, by default, "visit" the page.

Agree! A similar issue was brought up for the gc-accessibility-passport recently: https://github.com/GCTC-NTGC/gc-accessibility-passport/issues/105

**Comment by tristan-orourke** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/GC-Digital-Talent-6092dc05157fa30010edaef9/issues/GCTC-NTGC/gc-digital-talent/2902?planning-poker) with ZenHub @petertgiles

**Comment by petertgiles** [comment]
I see 94 results to check in 41 files.
![image](https://user-images.githubusercontent.com/8978655/171626662-391eac51-448e-4a62-94fc-a0e864c3df83.png)
